src/test/statistics/statistics_import_export_test.cpp

```diff
@@ -0,0 +1,44 @@
+#include <fstream>
+
+#include "gtest/gtest.h"
+
+#include "base_test.hpp"
+#include "statistics/column_statistics.hpp"
+#include "statistics/statistics_import_export.hpp"
+#include "statistics/table_statistics.hpp"
+#include "testing_assert.hpp"
+#include "utils/load_table.hpp"
+
+namespace opossum {
+
+class StatisticsImportExportTest : public ::testing::Test {};
+
+TEST_F(StatisticsImportExportTest, EndToEnd) {
+  std::vector<std::shared_ptr<const BaseColumnStatistics>> original_column_statistics;
+
+  original_column_statistics.emplace_back(std::make_shared<ColumnStatistics<int32_t>>(0.3f, 50.1f, 21, 100));
+  original_column_statistics.emplace_back(std::make_shared<ColumnStatistics<int64_t>>(0.4f, 51.2f, 22, 101));
+  original_column_statistics.emplace_back(std::make_shared<ColumnStatistics<float>>(0.5f, 51.3f, 2.2f, 1.01f));
+  original_column_statistics.emplace_back(std::make_shared<ColumnStatistics<double>>(0.6f, 52.3f, 2.2444, 1.01555));
+  original_column_statistics.emplace_back(std::make_shared<ColumnStatistics<std::string>>(0.7f, 53.3f, "abc", "xyz"));
+
+  TableStatistics original_table_statistics{TableType::Data, 3500, original_column_statistics};
+
+  const auto exported_statistics_file_path = test_data_path + "exported_table_statistics.json";
+
+  export_table_statistics(original_table_statistics, exported_statistics_file_path);
+
+  const auto imported_table_statistics = import_table_statistics(exported_statistics_file_path);
+
+  EXPECT_EQ(imported_table_statistics.table_type(), TableType::Data);
+  EXPECT_EQ(imported_table_statistics.row_count(), 3500);
+  ASSERT_EQ(imported_table_statistics.column_statistics().size(), 5u);
+
+  EXPECT_INT32_COLUMN_STATISTICS(imported_table_statistics.column_statistics().at(0), 0.3f, 50.1f, 21, 100);
```

Comment:
Both the constructor interface of `ColumnStatistics` and the one of the expect are not understandable from the outside. On the one hand, we add `const UseMvcc use_mvcc = UseMvcc::No` to avoid a bool argument for the table, on the other, we have a list of random numbers here. :(